sos/plugins/crash_report.py

```diff
@@ -0,0 +1,249 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+import os
+import sys
+import time
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+from sos.utilities import sos_get_command_output, find
+from datetime import datetime
+
+
+class CrashLog(Plugin):
+    """Crash kernel report
+    """
+    plugin_name = "crash_report"
+    profiles = ('system', 'debug')
+
+    dbg_vmlinux_path = '/usr/lib/debug/'
+    vmcore_stnd_path = '/var/crash'
+
+    dump_file_prefix = ''
+    crashed_kernel_name = ''
+    vmlinux_path = ''
+    message_to_log = ''
+    vmcore_file_path = ''
+    crash_script = ""
+    # crash result output file
+    crashed_output_file = '/var/log/os_crash_report.txt'
+
+    # list of crash command
+    crash_cmd_list = [
+                    ("sys", "\"SYSTEM INFORMATION:\""),
+                    ("mach", "\"MACHINE SPECIFIC DATA:\""),
+                    ("mach", "\"MACHINE SPECIFIC DATA:\""),
+                    ("bt", "\"STACK TRACE OF CURRENT CONTEXT:\""),
+                    ("bt -a", "\"STACK TRACES OF ACTIVE TASKS:\""),
+                    ("kmem -i", "\"MEMORY USAGE:\""),
+                    ("kmem -s", "\"KMALLOC SLAB DATA:\""),
```

Comment:
Is there a concern on large memory systems about how long any kmem commands take?  I know upstream crash did some improvements when walking SLAB tables so this would be better than it used to be, but I'd suspect anyone who's done core analysis on a system that had a large SLAB table or some type of memory-related issue knows how long `kmem` can take when it is trying to wrestle with huge indexes.  Two examples, an ideal and a problematic one:

```bash
$ cat crash_commands 
sys
mach
bt
bt -a
kmem -i
kmem -s
mod
ps
log
files
dev -p
runq
mach -o
irq -s
exit

$ time crash -i crash_commands vmcore /usr/lib/debug/lib/modules/2.6.32-358.el6.x86_64/vmlinux -s
- - - - - - - - 8< - - - - - - - - 
real	0m7.662s
user	0m7.343s
sys	0m0.302s
```

The above core had 16GB of ram and 2 CPU's so that seems a reasonable amount of time to me.  But depending on the issue this can take a very long time, such as any core hitting an [anon_vma_chain](https://lkml.org/lkml/2011/6/17/3) issue:

```bash
$ time crash -i crash_commands vmcore /usr/lib/debug/lib/modules/2.6.32-131.21.1.el6.x86_64/vmlinux -s
- - - - - - - - 8< - - - - - - - - 
real	23m28.618s
user	2m3.791s
sys	0m18.660s
```

The above was with a core with 250GB of RAM and 32 CPU's, with a massive amount of SLAB objects in the table due to the aforementioned issue - and it didn't finish, I actually killed it.  The vast majority of time was in `kmem -s` as it indexed the SLAB table and it would have eventually completed but probably only after an hour or so, myabe more.

So I have a real concern about how long this could take, or if we have a way to bail out if it takes an inordinate amount of time.  I'm playing devil's advocate here with a worst-case scenario but I still think it's a real possibility that would prevent `sosreport` from being effective if this was a default plugin.